avm/ptn/sa/conversation-knowledge-mining/main.bicep

```diff
@@ -0,0 +1,1395 @@
+
+  @description('Optional. If set to true, The Key Vault soft delete will be enabled.')
+  softDeleteEnabled: bool?
+
+  @description('Optional. The number of days to retain the soft deleted vault. If empty, it will be set to 7.')
+  @minValue(7)
+  @maxValue(90)
+  softDeleteRetentionInDays: int?
+
+  @description('Optional. If set to true, The Key Vault purge protection will be enabled. If empty, it will be set to false.')
+  purgeProtectionEnabled: bool?
+
+  @description('Optional. Array of role assignments to include in the Key Vault.')
+  roleAssignments: roleAssignmentType[]?
+}
+
+@export()
+@description('The type for the Conversation Knowledge Mining Log Analytics Workspace resource configuration.')
+type ckmLogAnalyticsWorkspaceType = {
+  @description('Optional. The name of the Log Analytics Workspace resource.')
+  @maxLength(63)
+  name: string?
+
+  @description('Optional. Location for the Log Analytics Workspace resource.')
+  @metadata({ azd: { type: 'location' } })
+  location: string?
+
+  @description('Optional. The SKU for the Log Analytics Workspace resource.')
+  sku: ('CapacityReservation' | 'Free' | 'LACluster' | 'PerGB2018' | 'PerNode' | 'Premium' | 'Standalone' | 'Standard')?
+
+  @description('Optional. The number of days to retain the data in the Log Analytics Workspace. If empty, it will be set to 30 days.')
+  @minValue(0)
+  @maxValue(730)
+  dataRetentionInDays: int?
+}
+
+@export()
+@description('The type for the Conversation Knowledge Mining Managed Identity resource configuration.')
+type ckmManagedIdentityType = {
+  @description('Optional. The name of the Managed Identity resource.')
+  @maxLength(128)
+  name: string?
+
+  @description('Optional. Location for the Managed Identity resource.')
+  @metadata({ azd: { type: 'location' } })
+  location: string?
+}
+
+@export()
+@description('The type for the Conversation Knowledge Mining Script resource configuration.')
+type ckmScriptType = {
+  @description('Optional. The name of the Script resource.')
+  @maxLength(90)
+  name: string?
+
+  @description('Optional. Location for the Script resource.')
+  @metadata({ azd: { type: 'location' } })
+  location: string?
+
+  @description('Optional. The base Raw Url of the GitHub repository where the Copy Data Script is located.')
+  githubBaseUrl: string?
+
+  @description('Optional. The Url where the Copy Data Script is located.')
+  scriptUrl: string?
+}
+
+@export()
+@description('The type for the Conversation Knowledge Mining SQL Server resource configuration.')
+type ckmSqlServerType = {
+  @description('Optional. The name of the SQL Server resource.')
+  @maxLength(63)
+  name: string?
+
+  @description('Optional. Location for the SQL Server resource.')
+  @metadata({ azd: { type: 'location' } })
+  location: string?
+
+  @description('Optional. The administrator login credential for the SQL Server.')
+  @secure()
+  #disable-next-line secure-parameter-default
+  sqlServerAdministratorLogin: string?
```

Comment:
Just an idea: For the properties that are already nested in a parameter that contains part of their name, you could leave that part out of the property itself.
What I mean is, the parameter is called `sqlServerConfiguration` - so instead of also calling the property `sqlServerAdministratorLogin` you could just call it `administratorLogin` as it's anyways clear it's for the SQL Server. Applies to other properties as well and would go a long way as it makes the parameters a shorter without loosing anything :)

Reply:
Updated properties for `ckmSqlServerType `. I left the properties for `ckmWebAppServerFarmType`, they refer to the configuration for the deployed web app:
```
param webAppServerFarmConfiguration ckmWebAppServerFarmType = {
  name: '${solutionPrefix}-wsrv'
  location: solutionLocation
  sku: 'B2'
  webAppResourceName: '${solutionPrefix}-app'
  webAppLocation: solutionLocation
  webAppDockerImageContainerRegistryUrl: 'kmcontainerreg.azurecr.io'
  webAppDockerImageName: 'km-app'
  webAppDockerImageTag: 'latest_2025-03-20_276'
}
```